app/src/lib/stores/app-store.ts

```diff
@@ -5031,23 +5041,29 @@ export class AppStore extends TypedBaseStore<IAppState> {
       return
     }
 
-    // get the previous stash before we create a new one
-    const previousStash = removePreviousStash
-      ? await getLastDesktopStashEntryForBranch(repository, branchName)
-      : null
+    if (preAction != null) {
+      const previousStashEntry = await getLastDesktopStashEntryForBranch(
+        repository,
+        branchName
+      )
+      await preAction(previousStashEntry)
+    }
 
     await createDesktopStashEntry(repository, branchName)
 
-    if (previousStash === null) {
-      return
-    }
+    if (postAction != null) {
+      const previousStashEntry = await getLastDesktopStashEntryForBranch(
+        repository,
+        branchName
+      )
 
-    await dropDesktopStashEntry(repository, previousStash.stashSha)
-    log.info(
-      `Dropped stash '${previousStash.stashSha}' associated with ${
-        previousStash.branchName
-      }`
-    )
+      // if we can't find the stash we just created, something went wrong
+      if (previousStashEntry === null) {
+        throw Error('Something bad happened')
```

Comment:
Any suggestions on useful info to put in this message would be helpful. This is a truly exceptional case and I don't know what getting `null` would mean.

Reply:
```suggestion
        throw new Error('The stash was not created. Check the logs for an underlying Git error because this shouldn't happen')
```